src/Clock.tsx

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-console */
+import React from 'react';
+
+type Props = {
+  clockName: string,
+};
+
+type State = {
+  date: Date;
+};
+
+export class Clock extends React.Component<Props, State> {
+  timerId = 0;
+
+  state = {
+    date: new Date(),
+  };
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ date: new Date() });
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    console.log(
+      `Old name: ${prevProps.clockName} / New name:${this.props.clockName}`,
+    );
```

Comment:
You don't need to print this msg every time the component updates, only if old name !== new name